src/lib.rs

```diff
@@ -286,30 +281,33 @@ impl Default for RequestParams {
 }
 
 /// Get a default `Client` and `RequestParams`.
-pub fn default() -> (hyper::Client, RequestParams) {
-    (hyper::Client::new(), RequestParams::default())
+pub fn default() -> (reqwest::Client, RequestParams) {
+    //FIXME: Bad in libs
+    let client = reqwest::Client::new().unwrap();
+    (client, RequestParams::default())
 }
 
 macro_rules! req_with_body {
     ($client:ident, $url:ident, $req:ident, $params:ident, $method:ident) => ({
     	let body = $req.body.expect("Expected this request to have a body. This is a bug, please file an issue on GitHub.");
 
 		let body: &[u8] = (*body).borrow();
-		let mut cursor = Cursor::new(body);
+//FIXME: My current understanding of copying, etc. doesn't get me far enough to grok this
```

Comment:
Ok so I've had a play with this and unfortunately that won't work in practice :( Because the body needs to be valid for the `'static` lifetime so it can be sent across threads. So after all that we're back to just copying the body again for now.

I'd say leave the method as you've got it, because I've got an idea to work around this using the macro. I'll open a new PR after getting your changes in.